Implemented time uniform for standalone animations

faazuthnar opened the pull request:
This is useful for animating stuff not depending on audio for additional visualizations.

This closes my issue: #120

coderobe commented:
Hmm, could this deprecate `tcounter` which is currently only used to figure out when a second has passed? Given that the semantics of both are similar, minus the 1-second-reset

faazuthnar commented:
Now, I removed `time`, which was a confusing name. Instead I made `fcounter` not reset every second, and added `fcounter_last`, which stores the last frame since the last second or so. I think it makes more sense this way.

coderobe commented:
`fcounter` is a frame counter, and thus varies by framerate and framerate stability - what's the use of using this as a measure of time, when it will be inconsistent without vsync (and even with vsync under adaptive refresh scenarios potentially)?

You're also casting `fcounter` (int) to `GLfloat` now, that can't be right.

coderobe commented:
Now i also just noticed that your initial commit had similar semantics regarding counting frames instead of realtime. I thought you wanted a measure of time passed?
While framecounters can be used to advance animations & scenes, you will end up with speedups/slowdowns relative to your reference speed on other devices.  I think you want a realtime counter.

faazuthnar commented:
Alright, I was rushed, I'll look into it further. Thanks

faazuthnar commented on glava/render.c:
```diff
@@ -2073,22 +2073,22 @@ bool rd_update(struct renderer* r, float* lb, float* rb, size_t bsz, bool modifi
+        ++gl->ucounter;               /*   increment update frame counter                 */
+        gl->kcounter = 0;             /*   reset keyframe counter (for interpolation)     */
+    } else ++gl->kcounter;            /* increment keyframe counter otherwise             */
+    gl->tcounter += (float) duration; /* timer counter, measuring when a >1s has occurred */
```
I'm not sure if this double to float cast is okay, but I didn't find any problems with it while debugging.

faazuthnar commented on glava/render.c:
```diff
@@ -125,8 +125,8 @@ struct gl_data {
     int rate; /* framerate */
-    double tcounter;
-    int fcounter, fcounter_last, ucounter, kcounter;
+    float tcounter, tcounter_last;
```
I made `tcounter` float, because double values are slower to compute with and require a GL extension on the shader side if I'm correct, so I decided on this.

faazuthnar commented:
Okay, now it seems more acceptable. Only thing I'm not keen on is that cast if you see the line comment.
